packages/grid/x-data-grid/src/hooks/features/pagination/useGridPagination.ts

```diff
@@ -1,41 +1,78 @@
 import * as React from 'react';
-import {
-  useGridPageSize,
-  defaultPageSize,
-  throwIfPageSizeExceedsTheLimit,
-} from './useGridPageSize';
 import { GridPrivateApiCommunity } from '../../../models/api/gridApiCommunity';
 import { DataGridProcessedProps } from '../../../models/props/DataGridProps';
-import { useGridPage, getPageCount } from './useGridPage';
 import { GridStateInitializer } from '../../utils/useGridInitializeState';
+import { GridPaginationApi, GridPaginationState } from './gridPaginationInterfaces';
+import { GridEventListener } from '../../../models/events';
+import { GridPaginationModel } from '../../../models/gridPaginationProps';
+import { gridVisibleTopLevelRowCountSelector } from '../filter';
+import { gridDensityFactorSelector } from '../density';
+import {
+  useGridLogger,
+  useGridSelector,
+  useGridApiMethod,
+  useGridApiEventHandler,
+} from '../../utils';
+import { GridPipeProcessor, useGridRegisterPipeProcessor } from '../../core/pipeProcessing';
+import { gridPaginationModelSelector } from './gridPaginationSelector';
+import { calculatePinnedRowsHeight } from '../rows/gridRowsUtils';
+import {
+  getPageCount,
+  noRowCountInServerMode,
+  defaultPageSize,
+  throwIfPageSizeExceedsTheLimit,
+  getDefaultGridPaginationModel,
+  getValidPage,
+} from './gridPaginationUtils';
 
 export const paginationStateInitializer: GridStateInitializer<
   Pick<
     DataGridProcessedProps,
-    'page' | 'pageSize' | 'rowCount' | 'initialState' | 'autoPageSize' | 'signature'
+    'paginationModel' | 'rowCount' | 'initialState' | 'autoPageSize' | 'signature'
   >
 > = (state, props) => {
-  let pageSize: number;
-  if (props.pageSize != null) {
-    pageSize = props.pageSize;
-  } else if (props.initialState?.pagination?.pageSize != null) {
-    pageSize = props.initialState.pagination.pageSize;
-    throwIfPageSizeExceedsTheLimit(pageSize, props.signature);
-  } else {
-    pageSize = defaultPageSize(props.autoPageSize);
-  }
+  const paginationModel =
+    (props.paginationModel ?? props.initialState?.pagination?.paginationModel) ||
+    getDefaultGridPaginationModel(props.autoPageSize);
```

Comment:
To allows partial specification in `initialState`
```suggestion
  const paginationModel = {
    ...getDefaultGridPaginationModel(props.autoPageSize),
    ...(props.paginationModel ?? props.initialState?.pagination?.paginationModel)
  };
```